packages/web3-eth-ens/src/contracts/Registry.js

```diff
@@ -38,7 +39,10 @@ function Registry(ens) {
     var self = this;
     this.ens = ens;
     this.contract = ens.checkNetwork().then(function (address) {
-        return new self.ens.eth.Contract(REGISTRY_ABI, address);
+        var contract = new Contract(REGISTRY_ABI, address);
+        contract.setProvider(self.ens.eth.currentProvider);
```

Comment:
Those lines can get removed as explained 1-2 times because it was only required because the provider inheritance didn't work and this will bring up the issue we had with the listener in the last review PR from you.